### DISCUSSION TITLE
Public Sponsor vs GitHub Sponsor

### DISCUSSION THREAD

**Discussion by pingshunhuangalex** [Schweinepriester/github-profile-achievements#16] [Ideas]
It's worth noting that previous sponsorship before `2022-06-09` won't be picked up by the new badges.

I've sponsored before and got the old `GitHub Sponsor`, but it was sneakily taken off after a while. Previously I thought this old badge might have an expiry date. Now come to think of it, it might be the GitHub team prepping for the new badges releases.

Sponsor a different project after `2022-06-09` helped me regain the new `Public Sponsor` badge with a slightly new design. I will observe this time more closely to see if it has an expiry date (I did a one-off sponsorship, not a monthly one)

**Comment by lnugraha** [comment]
I am not sure if my response answers correctly, but somehow user can lose their sponsorship badge (in the following month) if they stop sponsoring. CMIW.

**Comment by 5HT2** [comment]
Tangentially, "Pull Shark" counts old pull requests, but "YOLO" does not count old pull requests, and is only given if the requirements are met since it was added.

**Reply by Schweinepriester** [reply]
While I feel like the tendency is correct

> "YOLO" does not count old pull requests, and is only given if the requirements are met since it was added.

isn't true as is.

E.g. @Erol444 was awarded it dating prior to being (publicly) added on 2022-06-09:

>  Unlocked 2 Mar

See https://github.com/Erol444?achievement=yolo&tab=achievements.

Curiously I just noted the merge was actually earlier even

> [Erol444](https://github.com/Erol444) merged commit [fa9ba62](https://github.com/luxonis/depthai-hardware/commit/fa9ba62e04a3b5676d61e1f573ecdd0cdc982a0c) into master [on 17 Feb](https://github.com/luxonis/depthai-hardware/pull/209#event-6089379947)

I wonder what's the oldest qualifying PR is we can find? 🤔 

Random totally unfounded theories:
- they added the code to check for it on 2022-03-02
- they only looked back a certain period of time - a month? 2022?

**Comment by pingshunhuangalex** [comment]
Confirmed that the sponsorship will badge will go away in a month if you stopped sponsoring.

**Reply by Schweinepriester** [reply]
thank you! :)

**Comment by pingshunhuangalex** [comment]
Update: Sponsor badge now becomes a permanent badge which won't get expired after 30 days (without a renewed subscription).

Checking the milestone, it'll only pick the very first sponsor made (including the time using the old version)

**Reply by Schweinepriester** [reply]
oh wow!

can you elaborate on

> including the time using the old version

the old version? you mean the previous badge?

**Reply by pingshunhuangalex** [reply]
Yes
